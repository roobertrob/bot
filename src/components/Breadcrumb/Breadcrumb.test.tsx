import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Breadcrumb } from '.';

const navigateTo = jest.fn();

describe('Breadcrumb component', () => {
  it('should render Breadcrumb', () => {
    render(
      <Breadcrumb
        navigateTo={navigateTo}
        items={[
          {
            children: 'Analise Geral',
            link: '/',
          },
          {
            children: 'Principal',
            link: '/',
          },
        ]}
      />,
    );
    expect(screen.getByTestId('breadcrumb-component')).toMatchSnapshot();

    expect(screen.getByTestId('breadcrumb-component')).toBeInTheDocument();

    expect(screen.getByTestId('breadcrumb-component')).toBeVisible();

    userEvent.click(screen.getByTestId('breadcrumb-first-item'));
    expect(navigateTo).toHaveBeenCalled();

    expect(
      screen.getByTestId('breadcrumb-last-item').className.includes('ml-4'),
    ).toBeTruthy();

    expect(
      screen.getByTestId('breadcrumb-first-item').className.includes('mr-4'),
    ).toBeTruthy();
  });
  it('should not click in the item without link', () => {
    render(
      <Breadcrumb
        navigateTo={navigateTo}
        items={[
          {
            children: 'Page 1',
            link: '/page-1',
          },
          {
            children: 'Page 2',
          },
        ]}
      />,
    );
    userEvent.click(screen.getByText('Page 2'));
    expect(navigateTo).not.toBeCalled();
  });
  it('should call function navigateTo once', () => {
    const { getByText } = render(
      <Breadcrumb
        navigateTo={navigateTo}
        items={[
          {
            children: 'Index',
            link: '/test',
          },
          {
            children: 'Page',
            link: '/test/page',
          },
          {
            children: 'Page 2',
            link: '/test/page/page-2',
          },
          {
            children: 'Current item',
          },
        ]}
        data-testid="breadcrumb"
      />,
    );

    userEvent.click(screen.getByText('Page 2'));
    expect(navigateTo).toHaveBeenCalledTimes(1);
  });
});
