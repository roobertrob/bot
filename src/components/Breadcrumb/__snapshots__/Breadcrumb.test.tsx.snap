// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should render Breadcrumb 1`] = `
<div
  class="rounded-lg bg-white h-[4.0625rem] flex items-center p-6 space-x-2"
  data-testid="breadcrumb-component"
>
  <nav>
    <ol
      class="flex items-center"
    >
      <li>
        <div
          class="flex items-center"
        >
          <button
            class="mr-4 font-bold text-body1"
            data-testid="breadcrumb-first-item"
            role="link"
          >
            <p
              class="space-x-3 "
            >
              Analise Geral
            </p>
          </button>
          <span
            aria-hidden="true"
            class="before:content-['â€¢'] text-font-color-disabled"
          >
            /
          </span>
        </div>
      </li>
      <li>
        <div
          class="flex items-center"
        >
          <span
            aria-current="page"
            class="ml-4 text-greenLight"
            data-testid="breadcrumb-last-item"
          >
            <p
              class="space-x-3"
            >
              Principal
            </p>
          </span>
        </div>
      </li>
    </ol>
  </nav>
</div>
`;
